lib/Dialect/FIRRTL/FIRRTLTypeInterfaces.cpp

```diff
@@ -0,0 +1,15 @@
+//===- FIRRTLTypeInterfaces.cpp - FIRRTL Type Interfaces --------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the type interfaces of the FIRRTL dialect.
```

Comment:
Thanks for the declares/describes/implements language!

Reply:
Stolen from HWTypeInterfaces :D